### DISCUSSION TITLE
Any plans for supporting multi-language on the macOS version?

### DISCUSSION THREAD

**Discussion by ghost** [MaaAssistantArknights/MaaAssistantArknights#6534] [Q&A] [answered]
Hello! I, speaking for the Korean Arknights community, would like to express my admiration for your MAA project first.

I currently use the MAA software on both Windows and macOS, but I found that the macOS version is only available in simplified Chinese. I was wondering if there are any plans for multilingual support. Or if you have limitations for the support, I want to make some help localizing it into Korean.

One more question: Will it be available for Reclamation Algorithm? Windows one has it, but the macOS version doesn't.

**Comment by Constrat** [comment, marked as answer]
As you can see from #6470 with the latest public beta (you can just update to the main release of 4.24.0) the Reclamation Algorithm was released. 

Now, for your other question of the multi-lingual adaptation.
As far as I know, there is no way to add a different language for now, as the Chinese texts are `hard coded` in the .swift files, unlike the Windows version, where we have [src/MaaWpfGui/Res/Localizations](https://github.com/MaaAssistantArknights/MaaAssistantArknights/tree/b8f62f3069d6e9393ad6d258a78b14af7249515f/src/MaaWpfGui/Res/Localizations) with the 5 different languages, but this is because the MaaCore is written to allow the XAML file usage.

**Reply by ghost** [reply]
I updated my MAA version and checked that the option for Reclamation Algorithm has been released.
And I did understand that multi-lingual supporting on the macOS is difficult than I think.

Appreciates for the answer.

**Reply by Constrat** [reply]
The client functions for overseas (kr, en, jp and tw) should work. What's missing is only the graphical language.

**Reply by ghost** [reply]
My one is already working on the Korean-localized client; I found the option for switching a server in the preferences.
Again, thanks for the answer and this incredible project.
